public/app/features/integrated-alerting/components/NotificationChannel/AddNotificationChannelModal/PagerDutyFields/PagerDutyFields.tsx

```diff
@@ -1,18 +1,38 @@
 import React, { FC } from 'react';
-import { TextInputField, validators } from '@percona/platform-core';
+import { RadioButtonGroupField, TextInputField, validators } from '@percona/platform-core';
 import { Messages } from '../AddNotificationChannelModal.messages';
 import { PagerDutyFieldsProps } from './PagerDutyFields.types';
+import { PagerDutyKeyType } from '../../NotificationChannel.types';
+import { PAGER_DUTY_TYPE_OPTIONS } from '../AddNotificationChannel.constants';
 
-export const PagerDutyFields: FC<PagerDutyFieldsProps> = ({ values }) => {
-  const { required } = validators;
-  const { routing, service } = values;
-  const routingValidators = service ? [] : [required];
-  const serviceValidators = routing ? [] : [required];
+const { required } = validators;
+const keyValidator = [required];
 
+export const PagerDutyFields: FC<PagerDutyFieldsProps> = ({ values, mutators }) => {
   return (
     <>
-      <TextInputField name="routing" label={Messages.fields.routingKey} validators={routingValidators} />
-      <TextInputField name="service" label={Messages.fields.serviceKey} validators={serviceValidators} />
+      <RadioButtonGroupField
+        inputProps={{
+          /**
+           * TODO We probably should change platform-core to allow to pass values to the options.
+           * That way, we can take the value from the event itself here and call the mutator, rather than checking the form values.
+           */
+          onInput: () => {
+            values?.keyType === PagerDutyKeyType.service
+              ? mutators?.resetKey(PagerDutyKeyType.routing)
+              : mutators?.resetKey(PagerDutyKeyType.service);
+          },
+        }}
+        name="keyType"
+        options={PAGER_DUTY_TYPE_OPTIONS}
+        initialValue={values?.keyType || PAGER_DUTY_TYPE_OPTIONS[0].value}
+        fullWidth
+      />
+      {values.keyType === PagerDutyKeyType.service ? (
+        <TextInputField name={PagerDutyKeyType.service} label={Messages.fields.serviceKey} validators={keyValidator} />
+      ) : (
+        <TextInputField name={PagerDutyKeyType.routing} label={Messages.fields.routingKey} validators={keyValidator} />
+      )}
```

Comment:
[Switch on type is a code smell](https://wiki.c2.com/?SwitchStatementsSmell), I'd use an object to map the type to the component to be returned

Reply:
@nicolalamacchia I wouldn't bother if it's only two fields. And it doesn't look like code smell here - it's called conditional rendering, which is widely used in react.

Reply:
I know about conditional rendering :) I'm arguing with the fact that there's a condition on a type here, which is usually not a good idea, furthermore only a couple of attributes are actually changing, so it's possible to simplify that even more.

However this is not a big deal, so I'm ok with leaving it as it is.

Reply:
LOL, I knew you know all and everything about conditional rendering, just checking )))

Reply:
I mean, it's explicit. If you put in an object, it will no longer be - you'll jump through the hoops :)

Reply:
Yep, I get your point, totally agree, it should be explicit

Reply:
@nicolalamacchia  I agree with you on the attributes there, it can simplified :))